File: backend/insurance-applications/src/main/java/com/stancloud/insuranceapplications/response/CustomException.java
package com.stancloud.insuranceapplications.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.http.HttpStatus;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
public class CustomException extends RuntimeException {
  private HttpStatus httpStatus;
  private String message;

  public static CustomException notFound(String message) {
    return new CustomException(HttpStatus.NOT_FOUND, message);
  }
  public static CustomException badRequest(String message) {
    return new CustomException(HttpStatus.BAD_REQUEST, message);
  }
}

File: backend/authentication-service/src/main/kotlin/com/stancloud/authenticationservice/response/CustomException.kt
package com.stancloud.authenticationservice.response

import org.springframework.http.HttpStatus

class CustomException(
  val httpStatus: HttpStatus = HttpStatus.INTERNAL_SERVER_ERROR,
  val errorMessage: String? = null,
  val errors: List<String> = listOf(),
) : RuntimeException() {
  companion object {
    private const val serialVersionUID = 1L
  }

}